src/Board.py

```diff
@@ -35,3 +36,243 @@ def __init__(self, style="default", size=5) -> None:
                 raise ValueError("There is no default board for that number of players")
         else:
             raise ValueError("Un-recognised map style")
+
+    def get(self, x, y):
+        return self.data[y][x]
+
+    def getDirectlyAdj(self, start=(0, 0), TerainFilter=None) -> list:
+        adjacentTileList = []
+        if start[1] > 0:
+            if start[1] % 2 == 0:  # for even rows
+                if (
+                    TerainFilter is None
+                    or TerainFilter == self.get(start[0], start[1] - 1).terrain
+                ):
```

Comment:
This check gets repeated a lot. How about we pull it to the bottom of the function. Something like:
```python
from typing import Sequence, List

def get_directly_adjacent(..., terrain_filter: Sequence[Terrain] | None) -> List[tuple[int, int]]:
    ...

    filtered_terrain = [(x, y) for x, y in adjacent_tile_list if self.get(x, y).terrain not in terrain_filter]]
```